kvbc/include/v4blockchain/v4_blockchain.h

```diff
@@ -136,6 +125,105 @@ class KeyValueBlockchain {
   // Precondition3: `block_id_at_checkpoint` <= getLastReachableBlockId()
   void trimBlocksFromSnapshot(BlockId block_id_at_checkpoint);
 
+  //////////////////Recovery infra///////////////////////////
+  /*
+  Recovery is needed when we crash while executing a BFT sequence number's requests or when trimming the blocks that
+  were added duting a checkpoint creation.
```

Comment:
Another que is, if the recovery process fails to create the recovery DB once, will it be possible to recover by restarting the replica. Currently, if we compare the recovery with categorized BC we will see that we can revert to a stable position even if there are crashes at the startup. So do we loose this property with this form of startup.

Reply:
if the metadata is cleared then the replica is considered to be in sync by the replica state sync and no blocks will be deleted.

the flow is resilient to crashes, let's go over it:
1- the recovery of requests and checkpoint will not use the same DB , the first will have path of “$rocksdbpath/recover_execution” and the other “$rocksdbpath/recover_chekpoint”
2 - When recovery is needed open an instance of the recover database.
3 - fill it with recovery data, do not override existing entries.
4 - now we open the blockchain in write mode
5 - delete blocks from the blockchain with recovery data
6 - on finish - delete the recovery instance
crashes:
if we crash in 3 - it will do it again, on next restart
if we crash in 4 - the write open is destructive, but the data is saved and will not be overwritten by 3.
if we crash in 5 - the data is still saved and we’ll not overwrite in 3 and it will continue from the last block id before the crash at 5.
if the deletion is successful we must call 6 and it guarantees that no garbage or stale data is left or used.